rflx/specification/parser.py

```diff
@@ -75,7 +76,7 @@ def type_location(identifier: ID, node: RFLXNode) -> Location:
 
 
 def diagnostics_to_error(
-    diagnostics: List[Diagnostic], error: RecordFluxError, specfile: Path = None
+    diagnostics: List[Diagnostic], error: RecordFluxError, filename: Path = STDIN
```

Comment:
IMHO it is better to make `filename` mandatory here:

```diff
diff --git a/rflx/specification/parser.py b/rflx/specification/parser.py
index b38674ab..9b896395 100644
--- a/rflx/specification/parser.py
+++ b/rflx/specification/parser.py
@@ -76,7 +76,7 @@ def type_location(identifier: ID, node: RFLXNode) -> Location:
 
 
 def diagnostics_to_error(
-    diagnostics: List[Diagnostic], error: RecordFluxError, filename: Path = STDIN
+    diagnostics: List[Diagnostic], error: RecordFluxError, filename: Path
 ) -> bool:
     """
     Append langkit diagnostics to RecordFlux error. Return True if error occured.
@@ -1189,7 +1189,7 @@ class Parser:
         self.__cache = Cache(cached)
 
     def __convert_unit(
-        self, spec: Specification, filename: Path = STDIN, transitions: List[ID] = None
+        self, spec: Specification, filename: Path, transitions: List[ID] = None
     ) -> RecordFluxError:
         transitions = transitions or []
         error = RecordFluxError()
@@ -1256,8 +1256,8 @@ class Parser:
     ) -> None:
         error = RecordFluxError()
         unit = AnalysisContext().get_from_buffer("<stdin>", string, rule=rule)
-        if not diagnostics_to_error(unit.diagnostics, error):
-            error = self.__convert_unit(unit.root)
+        if not diagnostics_to_error(unit.diagnostics, error, STDIN):
+            error = self.__convert_unit(unit.root, STDIN)
             for f, (packagefile, spec) in self.__specifications.items():
                 check_naming(error, spec.f_package_declaration, f, packagefile)
         error.propagate()
```

Reply:
Changed in ade57632